src/config/externalPaths.ts

```diff
@@ -9,8 +11,14 @@ import { buildItemPath } from './paths';
 
 export const buildGraaspPlayerView = (id: string): string =>
   `${GRAASP_PLAYER_HOST}/${id}`;
-export const buildGraaspBuilderView = (id: string): string =>
-  `${window.location.origin}${buildItemPath(id)}`;
+export const buildGraaspBuilderView = (
+  id: string,
+  mode = DEFAULT_ITEM_LAYOUT_MODE,
+): string => {
+  const url = new URL(`${window.location.origin}${buildItemPath(id)}`);
```

Comment:
You can make use of the `base` argument of `URL`:
```suggestion
  const url = new URL(buildItemPath(id), window.location.origin);
```